### DISCUSSION TITLE
Releasing the audio stream

### DISCUSSION THREAD

**Discussion by taste-kitchen** [samhirtarif/react-audio-recorder#74] [Q&A] [answered]
This is probably a daft question but currently I am using the useAudioRecorder hook in a conditionally rendered component. But I am noticing the stream doesn't get released until I refresh the entire tab. Can anyone help me out as to how I can clean up efficiently in useEffect?

**Comment by samhirtarif** [comment]
What version of the package are you using? This was something that I had fixed in the very initial stages of this package's development

**Reply by taste-kitchen** [reply]
I'm on the latest 2.1.2 so i suspect it's something to do with my implementation?

**Reply by samhirtarif** [reply]
Would be really helpful if you could share a minimal reproduction of the issue on stackblitz or codepen

**Reply by taste-kitchen** [reply, marked as answer]
Hi! I found the error with my implementation (I wasn't cleaning up the component properly).
